flocker/node/_docker.py

```diff
@@ -527,6 +528,50 @@ def _address_in_use(self, apierror):
             return None
         return AddressInUse(address=(ip, int(port)), apierror=apierror)
 
+    def _image_data(self, image, container, state):
+        """
+        Supply data about an image, by either inspecting it or returning
+        cached data if available.
+
+        :param unicode image: The ID of the image.
+
+        :param unicode container: The ID of the container for which this
+            image is being used.
+
+        :param unicode state: The state of the container.
+
+        :return: ``dict`` representing data about the image properties.
+        """
+        # This caching technique isn't thread-safe, but allowable
+        # in this case because a) the data is for our purposes
+        # immutable, so two writes of same key isn't a problem and
+        # b) we never remove entries.
+        if image in self._image_cache:
+            return self._image_cache[image]
+        try:
+            image_data = self._client.inspect_image(image)
+            self._image_cache[image] = {
+                u"Config": {
+                    u"Cmd": image_data[u"Config"][u"Cmd"],
+                    u"Env": image_data[u"Config"][u"Env"],
+                }
+            }
+        except APIError as e:
```

Comment:
Ideally only the code that might actually raise the exception being handled should be inside the try block.  The second statement, the statement populating the cache, should be outside the try block, near the `return image_data`.